s2cloudless/pixel_classifier.py

```diff
@@ -30,16 +30,15 @@ def __init__(self, classifier: Any):
         """
         :param classifier: An instance of trained classifier that will be executed over an entire image
         """
-        self._check_classifier(classifier)
-        self.classifier = classifier
+        self.classifier = self._check_classifier(classifier)
```

Comment:
not a fan of having `_check_classifier` return a classifier, it doesn't suit the name.
as for the rest, this way you actually have one extra line in the code :)

Reply:
reverted and updated, can not remove Booster check in _check_classifier